lib/Command/IndexPageContentCommand.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\IbexaSearchExtra\Command;
+
+use Ibexa\Contracts\Core\Persistence\Handler as PersistenceHandler;
+use Ibexa\Contracts\Core\Repository\ContentService;
+use Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException;
+use Ibexa\Contracts\Core\Repository\Exceptions\NotFoundException;
+use Ibexa\Contracts\Core\Repository\Exceptions\UnauthorizedException;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion;
+use Ibexa\Contracts\Core\Repository\Values\Filter\Filter;
+use Ibexa\Contracts\Core\Search\Handler as SearchHandler;
+use Netgen\IbexaSearchExtra\Exception\IndexPageUnavailableException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Ibexa\Contracts\Core\Repository\Values\Content\ContentList;
+
+use function count;
+use function explode;
+
+class IndexPageContentCommand extends Command
+{
+    protected static $defaultName = 'netgen-search-extra:index-page-content';
+
+    /**
+     * @param ContentService $contentService
+     * @param SearchHandler $searchHandler
+     * @param PersistenceHandler $persistenceHandler
+     * @param array<string> $allowedContentTypes
+     */
+    public function __construct(
+        private readonly ContentService $contentService,
+        private readonly SearchHandler $searchHandler,
+        private readonly PersistenceHandler $persistenceHandler,
+        private readonly array $allowedContentTypes,
+    ) {
+        parent::__construct($this::$defaultName);
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setDescription('Index content related through layouts')
+            ->addOption(
+                'content-ids',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Comma separated list of content id\'s of content to index.',
+            );
+    }
+
+    /**
+     * @throws NotFoundException
+     * @throws InvalidArgumentException
+     * @throws UnauthorizedException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $contentIds = $input->getOption('content-ids');
+        if ($contentIds !== null) {
+            $contentIds = explode(',', $contentIds);
```

Comment:
Please extract this block and `else` block into separate private methods.

Reply:
resolved in https://github.com/netgen/ibexa-search-extra/commit/28c2082565f4b56880ebae582f4b8de43fedcae3